src/main/java/com/unstoppabledomains/resolution/naming/service/ENS.java

```diff
@@ -92,6 +96,30 @@ public String getOwner(String domain) throws NamingServiceException {
     throw new NamingServiceException(NSExceptionCode.NotImplemented, new NSExceptionParams("m|n", "getTokensOwnedBy", getType().toString()));
   }
 
+  @Override
+  public Map<String, Location> getLocations(String... domains) throws NamingServiceException {
+    Map<String, Location> locations = new HashMap<>();
+    for (String domain : domains) {
+      try {
+        Location location = new Location();
+        location.setBlockchain("ETH");
+        location.setBlockchainProviderURL(this.getProviderUrl());
+        location.setNetworkId(this.getNetwork());
+        location.setOwner(this.getOwner(domain));
+        location.setRegistryAddress(this.contractAddress);
+        location.setResolverAddress(this.getResolverAddress(domain));
+        locations.put(domain, location);
```

Comment:
We can technically get all the data for this method. And for a few domains the performance shouldn't be too bad. So I don't think it would be fair to say it is unsupported. We can always improve it later.

I can also change it to request data for each domain in parallel, in case anyone wants to get locations of a bunch of domains at the same time.

@DeRain what do you think?